miio/gateway.py

```diff
@@ -529,45 +581,295 @@ def set_light(self, color_name, brightness):
+
+    @property
+    def pressure(self):
+        """Return the pressure in hPa"""
+        return self._pressure
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(["temperature", "humidity", "pressure"])
+        self._temperature = values[0] / 100
+        self._humidity = values[1] / 100
+        self._pressure = values[2] / 100
+
 
 class SensorHT(SubDevice):
     accessor = "get_prop_sensor_ht"
     properties = ["temperature", "humidity"]
+    _temperature = None
+    _humidity = None
+
+    @property
+    def temperature(self):
+        """Return the temperature in degrees celsius"""
+        return self._temperature
+
+    @property
+    def humidity(self):
+        """Return the humidity in %"""
+        return self._humidity
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(self.properties)
+        self._temperature = values[0] / 100
+        self._humidity = values[1] / 100
 
 
-class Plug(SubDevice):
+class AqaraMagnet(SubDevice):
+    _status = None
+
+    @property
+    def status(self):
+        """Returns 'open' or 'closed'"""
+        return self._status
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(["unkown"])
+        self._status = values[0]
+
+
+class AqaraPlug(SubDevice):
     accessor = "get_prop_plug"
     properties = ["power", "neutral_0"]
+    _power = None
+    _status = None
+
+    @property
+    def power(self):
+        """Return the power consumption"""
+        return self._power
+
+    @property
+    def status(self):
+        """Return the status of the plug: on/off"""
+        return self._status
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(self.properties)
+        self._power = values[0]
+        self._status = values[1]
+
+
+class AqaraRelayTwoChannels(SubDevice):
+    _status_ch0 = None
+    _status_ch1 = None
+    _load_power = None
+
+    class AqaraRelayToggleValue(Enum):
+        """Options to control the relay"""
+
+        toggle = "toggle"
+        on = "on"
+        off = "off"
+
+    class AqaraRelayChannel(Enum):
+        """Options to select wich relay to control"""
+
+        first = "channel_0"
+        second = "channel_1"
+
+    @property
+    def status_ch0(self):
+        """Return the state of channel 0"""
+        return self._status_ch0
+
+    @property
+    def status_ch1(self):
+        """Return the state of channel 1"""
+        return self._status_ch1
+
+    @property
+    def load_power(self):
+        """Return the load power"""
+        return self._load_power
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(["load_power", "channel_0", "channel_1"])
```

Comment:
Define the property names as class attributes (like other subdevices are already doing).

Reply:
done